src/main/java/seedu/address/logic/parser/FacilCommandParser.java

```diff
@@ -0,0 +1,179 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE_CODE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_OFFICE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+
+import java.util.*;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.facil.*;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.facilitator.*;
+import seedu.address.model.module.ModuleCode;
+
+/**
+ * Parses input arguments and creates a new FacilAdd object
+ */
+public class FacilCommandParser implements Parser<FacilCommand> {
+
+    public FacilCommand parse(String args, String commandWord) throws ParseException {
+        switch (commandWord) {
+            case FacilCommand.ADD_COMMAND_WORD:
+                return parseAdd(args);
+            case FacilCommand.EDIT_COMMAND_WORD:
+                return parseEdit(args);
+            case FacilCommand.DELETE_COMMAND_WORD:
+                return parseDelete(args);
+            case FacilCommand.FIND_COMMAND_WORD:
+                return parseFind(args);
+            default:
+                throw new ParseException("Error");
+        }
```

Comment:
The indentation. and maybe throw new ParseException(MESSAGE_UNKNOWN_FACILITATOR_COMMAND)